deltachat-ffi/src/lib.rs

```diff
@@ -1079,30 +1088,41 @@ pub type dc_chatlist_t<'a> = dc_chatlist::dc_chatlist_t<'a>;
 
 #[no_mangle]
 pub unsafe extern "C" fn dc_chatlist_unref(chatlist: *mut dc_chatlist::dc_chatlist_t) {
-    dc_chatlist::dc_chatlist_unref(chatlist)
+    assert!(!chatlist.is_null());
+
+    let _list = &*chatlist;
```

Comment:
It does not move the vaule, so it is not an equivalent to `std::mem::drop`.

I am still not sure what it does, seems like a no-op. It doesn't even prevent compiler warnings about unused variable, since `chatlist` is already used in assertion.

Reply:
And since `dc_get_chatlist` returns `Box::into_raw(Box::new(list))`, I think this one should do `Box::from_raw(chatlist)`, otherwise we have a memory leak.

Reply:
should `from_raw` for dropping, you are right

Reply:
fixed